dokka-subprojects/analysis-kotlin-symbols/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/symbols/translators/DefaultSymbolToDocumentableTranslator.kt

```diff
@@ -109,11 +109,16 @@ internal class DokkaSymbolVisitor(
         get() = (psi as? KtModifierListOwner)?.hasExpectModifier() == true
 
     private fun <T : KtSymbol> Collection<T>.filterSymbolsInSourceSet() = filter {
-        it.psi?.containingFile?.virtualFile?.path?.let { path ->
-            path.isNotBlank() && sourceSet.sourceRoots.any { root ->
-                Paths.get(path).startsWith(root.toPath())
+        val pathString = it.psi?.containingFile?.virtualFile?.path
+        when {
+            pathString.isNullOrBlank() -> false
+            else -> {
+                val absolutePath = Paths.get(pathString).toAbsolutePath()
+                sourceSet.sourceRoots.any { root ->
+                    absolutePath.startsWith(root.toPath().toAbsolutePath())
```

Comment:
Not really, it works as expected, as this is not `String.startsWith(String)` but `Path.startsWith(Path)` - which does structural check

<img width="656" alt="image" src="https://github.com/Kotlin/dokka/assets/50462752/da035040-f785-4530-84f3-dc15c1837ad7">

Reply:
Sorry, I have not noticed it. 

But can you check how it works with a shorthand (e.g. `..`)? and symbolic links?

Reply:
So, yes, there is a problem that `toAbsolutePath` will not remove `..` and will not follow symlinks. To remove `..` we need to use `Path.normalize` and to follow symlinks and remove `..` we need to use `Path.toRealPath`.
Also, `Path.toRealPath` will fail if there will be some non-existent directory in the path (even if it will be removed because of `..`), while just `normalize`/`toAbsolutePath` - not

<img width="1019" alt="image" src="https://github.com/Kotlin/dokka/assets/50462752/33274378-eba3-4d8b-a798-19b6b5b0f9f2">

Reply:
I've replaced `Path.toAbsolutePath` with `Path.toRealPath` so symlinks and `..` will be fully resolved now.
And also, as I mentioned, execution will now fail if there is some non-existent path, like in a test fixed here: https://github.com/Kotlin/dokka/pull/3362/commits/519e9ffdff239faa8944317e8b2db291dc04853d

Reply:
Resolving symlinks can be excessive. 
I have tested: e.g. if `pathString = "/.../project/src/symlink/file.kt"` is resolved with a symlink,  `absolutePath` will lead to a symlink `/pathToSymLink/file.kt`.

I think it would be safer to use `analysisContext.modulesWithFiles[sourceModule]` in K2 for filtering. It can help to avoid problems with paths.
But it does not work in K1.